external/fv3fit/fv3fit/pytorch/graph/train.py

```diff
@@ -46,8 +46,8 @@ class GraphHyperparameters(Hyperparameters):
     optimizer_config: OptimizerConfig = dataclasses.field(
         default_factory=lambda: OptimizerConfig("AdamW")
     )
-    graph_network: UNetGraphNetworkConfig = dataclasses.field(
-        default_factory=lambda: UNetGraphNetworkConfig()
+    graph_network: MPGUNetGraphNetworkConfig = dataclasses.field(
```

Comment:
Your GraphHyperparameters class should support being able to select whether you want to produce a u-net or a message passing u-net. There are a few ways to do this, one way is to use the same configuration object for both and have it produce the right u-net based on the options the user set. Another is to make this type a Union of types (assuming they are different enough that dacite can tell which one is which).